src/Appwrite/Platform/Workers/Databases.php

```diff
@@ -248,113 +251,117 @@ private function deleteAttribute(Document $database, Document $collection, Docum
+                $orders = $index->getAttribute('orders');
+
+                $found = \array_search($key, $attributes);
+
+                if ($found !== false) {
+                    // If found, remove entry from attributes, lengths, and orders
+                    // array_values wraps array_diff to reindex array keys
+                    // when found attribute is removed from array
+                    $attributes = \array_values(\array_diff($attributes, [$attributes[$found]]));
+                    $lengths = \array_values(\array_diff($lengths, isset($lengths[$found]) ? [$lengths[$found]] : []));
+                    $orders = \array_values(\array_diff($orders, isset($orders[$found]) ? [$orders[$found]] : []));
+
+                    if (empty($attributes)) {
+                        $dbForProject->deleteDocument('indexes', $index->getId());
+                    } else {
+                        $index
+                            ->setAttribute('attributes', $attributes, Document::SET_TYPE_ASSIGN)
+                            ->setAttribute('lengths', $lengths, Document::SET_TYPE_ASSIGN)
+                            ->setAttribute('orders', $orders, Document::SET_TYPE_ASSIGN);
+
+                        // Check if an index exists with the same attributes and orders
+                        $exists = false;
+                        foreach ($indexes as $existing) {
+                            if (
+                                $existing->getAttribute('key') !== $index->getAttribute('key') // Ignore itself
+                                && $existing->getAttribute('attributes') === $index->getAttribute('attributes')
+                                && $existing->getAttribute('orders') === $index->getAttribute('orders')
+                            ) {
+                                $exists = true;
+                                break;
+                            }
                         }
-                    }
 
-                    if ($exists) { // Delete the duplicate if created, else update in db
-                        $this->deleteIndex($database, $collection, $index, $project, $dbForConsole, $dbForProject);
-                    } else {
-                        $dbForProject->updateDocument('indexes', $index->getId(), $index);
+                        if ($exists) { // Delete the duplicate if created, else update in db
+                            $this->deleteIndex($database, $collection, $index, $project, $dbForConsole, $dbForProject);
+                        } else {
+                            $dbForProject->updateDocument('indexes', $index->getId(), $index);
+                        }
                     }
                 }
             }
-        }
-
-        $dbForProject->purgeCachedDocument('database_' . $database->getInternalId(), $collectionId);
-        $dbForProject->purgeCachedCollection('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId());
+        } finally {
```

Comment:
Are we able to move this logic to the inner finally to avoid the nested try/catch like the other methods?